File: fazendas_front/src/types/fazenda.ts
export interface Fazenda {
  id: number
  cod_imovel?: string
  municipio?: string
  num_area?: number
  lat?: number
  lng?: number
}

export interface BuscaPontoRequest {
  latitude: number
  longitude: number
  page: number
  page_size: number
}

export interface BuscaRaioRequest {
  latitude: number
  longitude: number
  raio_km: number
  page: number
  page_size: number
}

File: fazendas_api/app/schemas/fazenda.py
from pydantic import BaseModel, Field
from typing import Optional


class Pagination(BaseModel):
    page: int = Field(1, ge=1)
    page_size: int = Field(20, ge=1, le=100)


class SearchPointRequest(Pagination):
    latitude: float
    longitude: float


class SearchRadiusRequest(Pagination):
    latitude: float
    longitude: float
    radius_km: float


class FarmResponse(BaseModel):
    id: int
    cod_imovel: Optional[str]
    municipio: Optional[str]
    num_area: Optional[float]

    class Config:
        orm_mode = True
